src/flags.rs

```diff
@@ -4,3 +4,4 @@ pub const FLAG_DOTNL: u32 = 1 << 2;
 pub const FLAG_SWAP_GREED: u32 = 1 << 3;
 pub const FLAG_IGNORE_SPACE: u32 = 1 << 4;
 pub const FLAG_UNICODE: u32 = 1 << 5;
+pub const FLAG_ONIG_MODE: u32 = 1 << 6;
```

Comment:
I've been thinking about this, and I realized that I'm not a huge fan of abbreviations, because they make it harder to search the codebase for something - i.e. if we search for `Oniguruma`, this won't show up. What do others think about using the full name? `FLAG_ONIGURUMA_MODE`  here and [`oniguruma_mode` in the `RegexOptions` struct](https://github.com/fancy-regex/fancy-regex/pull/186/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R552)?

Reply:
I'm totally fine with that! Feel free to push that change if you want. I won't be able to get to things myself till later next week